frontend/src/RecordingPageComp/PlaybackContainerComp.jsx

```diff
@@ -1,10 +1,58 @@
 /* eslint-disable react/prop-types */
+import { useEffect, useRef } from "react";
+import { PitchDetector } from "pitchy";
+
 function PlaybackContainer({
   midiSounds,
   note,
   playbackDuration,
   selectedInstrument,
+  selectedBlob,
+
+  setNote,
+  setPlaybackDuration,
 }) {
+  const audioContextRef = useRef(new AudioContext());
+
+  function frequencyToMIDINoteNumber(frequency) {
```

Comment:
The naming of the function implies it's only a utility function that performs and returns the frequency conversion. However, it sets the state variable after the conversion. I would suggest setting the state variable outside of this function or renaming the function to fully represent what the function is doing